onnxscript/rewriter/ort_fusions/sdpa.py

```diff
@@ -57,34 +58,55 @@ def check(self, op, query, key_transposed, value, mask, query_scale, key_scale,
 
         if self._pre_scale:
             # Check if query_scale and key_scale are scalars == sqrt(expected_scaling_factor)
+            # If they are scalars but != sqrt(expected_scaling_factor), a custom scale is being used.
             sqrt_scaling_factor = math.sqrt(expected_scaling_factor)
-            if not _ir_utils.is_singleton_value(query_scale, sqrt_scaling_factor, rtol=1e-3):
+            # Calculate the scaling factor for query
+            if _ir_utils.get_singleton_value(query_scale) is None:
                 return check_result.fail(
-                    "Query scale is not a scalar or does not match the expected scaling factor.",
+                    "Query scale is not a scalar.",
                     query_scale,
                 )
-            if not _ir_utils.is_singleton_value(key_scale, sqrt_scaling_factor, rtol=1e-3):
+            if not _ir_utils.is_singleton_value(query_scale, sqrt_scaling_factor, rtol=1e-3):
+                query_scale_value = _ir_utils.get_singleton_value(query_scale)
+                self._scale = query_scale_value * query_scale_value
+            else:
+                self._scale = expected_scaling_factor
+            # Ensure the scaling factor for key is the same as for query
+            if _ir_utils.get_singleton_value(key_scale) is None:
                 return check_result.fail(
-                    "Key scale is not a scalar or does not match the expected scaling factor.",
+                    "Key scale is not a scalar.",
                     key_scale,
                 )
+            if not _ir_utils.is_singleton_value(key_scale, sqrt_scaling_factor, rtol=1e-3):
+                if _ir_utils.get_singleton_value(query_scale) != _ir_utils.get_singleton_value(
```

Comment:
nit: this is a float-value comparison. I think some comparison with atol/rtol would be better (but we need some thought on what those values should be, even in other places where we use 1e-3 ... that's a generic pre-existing TODO.)

Reply:
Used a float-value comparison with rtol for now. Could fix/propose a mechanism to allow caller to set rtol/atol in a future PR as an extension.